### ISSUE TITLE
Never reprompted for new password, no keychain entry recreated

### ISSUE THREAD

**Issue by ghostsquad** [Versent/saml2aws#873]
running version 2.36.0 on OSX Monterey 12.5. I deleted `~/.saml2aws` and the entry in my OSX Keychain. Rerun `saml2aws configure`, and I'm not prompted for my password again.

```
❯ saml2aws login --skip-prompt --force --verbose
DEBU[0000] Running                                       command=login
DEBU[0000] Check if creds exist.                         command=login
DEBU[0000] Expand                                        name=/Users/<REDACTED>/.aws/credentials pkg=awsconfig
DEBU[0000] resolveSymlink                                name=/Users/<REDACTED>/.aws/credentials pkg=awsconfig
DEBU[0000] ensureConfigExists                            filename=/Users/<REDACTED>/.aws/credentials pkg=awsconfig
Using IdP Account default to access Okta https://<REDACTED>.okta.com/home/amazon_aws/<REDACTED>
DEBU[0000] building provider                             command=login idpAccount="account {\n  DisableSessions: false\n  DisableRememberDevice: false\n  URL: https://<REDACTED>.okta.com/home/amazon_aws/<REDACTED>\n  Username: <REDACTED>\n  Provider: Okta\n  MFA: TOTP\n  SkipVerify: false\n  AmazonWebservicesURN: urn:amazon:webservices\n  SessionDuration: 43200\n  Profile: saml\n  RoleARN: \n  Region: \n}"
DEBU[0000] okta | disableSessions: false                 provider=okta
DEBU[0000] okta | rememberDevice: true                   provider=okta
Empty password
Error validating login details.
github.com/versent/saml2aws/v2/cmd/saml2aws/commands.Login
	github.com/versent/saml2aws/v2/cmd/saml2aws/commands/login.go:85
main.main
	./main.go:188
runtime.main
	runtime/proc.go:250
runtime.goexit
	runtime/asm_arm64.s:1263
```

If I just run `saml2aws login`, I'm prompted for my password, and I everything works.

This issue is specifically that something about regenerating the OSX Keychain entry is broken.

**Comment by ghostsquad** [comment]
I have found that this isn't a problem if I install saml2aws with brew. But it is a problem if I install it via ASDF. Not sure what's different.

**Comment by QuentinBtd** [comment]
Hello,

I just see that today with a manual binary installation. Like @ghostsquad, no problem with brew. 

Both are in 2.

I don't know how work Keychain, if we have to allow the binary to access to it...

**Comment by QuentinBtd** [comment]
Hi

I have an answer ;

It works with the binary provided by Homebrew because it is signed with codesign.

The binary available on Github seems not signed. So that it can't access to keychain.

**Comment by QuentinBtd** [comment]
There is something weird ...

I built saml2aws without signing binary, and it works with keychain...

**Comment by QuentinBtd** [comment]
Okay; everything I wrote before is wrong.

But I've found something strange with `configure` option.

With the binary provided by Homebrew, of build by my, I have to enter my password two times ; 
![image](https://user-images.githubusercontent.com/21245565/189504042-4905b64e-d3a0-4d07-b95c-9f58daace48b.png)

With binaries downloaded on GitHub, it isn’t ask 

<img width="614" alt="Capture d’écran 2022-09-11 à 00 33 56" src="https://user-images.githubusercontent.com/21245565/189504096-a95f0ea7-89b5-48e3-97aa-1b6f78be8702.png">

If someone has an idea...

**Comment by QuentinBtd** [comment]
Okay, I got it. 

For information, I'm a Macbook Pro M1 user.

`saml2aws configure` ask password (and stored in Keychain) with there binaries: 
- installed by Homebrew
- builded myself (tested with Go 1.18 and 1.19)
- downloaded on GitHub in version for AMD platform

The password is not asked (and not stored to Keychain) only with versions for ARM platform downloaded here. 

Maybe something is wrong during the build ? @wolfeidau

**Comment by QuentinBtd** [comment]
I built the binary for `darwin arm64` platform on my amd64 laptop. 
I try the binary on my Mac... I have the problem. 

This seems to come from the fact that the `darwin arm64` binary is built on a server/PC with a amd64 platform CPU.

**Comment by elementalvoid** [comment]
👋 Experiencing the same here. This only happens with the M1/ARM build. I'm sorry for the length of this post, but I wanted to make sure all the relevant data is included.

I stopped using the asdf installation method because the "official" binary released on GH Releases doesn't support Keychain on M1. FWIW, I am the maintainer of the saml2aws asdf plugin; asdf users experience this issue because the plugin downloads the binary from the GH Releases page. However, Brew binaries are built differently so they continue to function.

>This seems to come from the fact that the darwin arm64 binary is built on a server/PC with a amd64 platform CPU.

I think the key part may be that the build is not performed on a Mac where it can link to the OSX Keychain libraries, or maybe that it's not detecting them. I believe a cross-compile should be fine so long as it can link to the Keychain lib.

Interestingly, this issue only occurs on the ARM builds. So, maybe cross-compiling is not okay?

---

Here are the inspection steps that I took to narrow it down to only Mac M1 (ARM).

I beleive it's due to shared library linking differences between the GH release binary and the Brew binary. At the end, we'll see that this only affects the ARM GH Release binary though; I don't know why.

---

Below, I install the latest GH Release binary. We see from the install log that it downloads the Darwin ARM64 build. Then, I check the file types of the two binaries and list the shared libraries for each.

Get latest GH Release binary
```
❯ asdf install saml2aws latest
Downloading saml2aws from https://github.com/Versent/saml2aws/releases/download/v2.36.0/saml2aws_2.36.0_Darwin_arm64.tar.gz
```

Check file types
```
❯ file /opt/homebrew/bin/saml2aws
/opt/homebrew/bin/saml2aws: Mach-O 64-bit executable arm64

❯ file /Users/matt.klich/.asdf/installs/saml2aws/2.36.0/bin/saml2aws
/Users/matt.klich/.asdf/installs/saml2aws/2.36.0/bin/saml2aws: Mach-O 64-bit executable arm64
```
Both are `Mach-O 64-bit executable arm64` which is desired.

Inspect shared libraries on GH Release binary:
```
❯ otool -L /Users/matt.klich/.asdf/installs/saml2aws/2.36.0/bin/saml2aws
/Users/matt.klich/.asdf/installs/saml2aws/2.36.0/bin/saml2aws:
        /usr/lib/libSystem.B.dylib (compatibility version 0.0.0, current version 0.0.0)
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 0.0.0, current version 0.0.0)
        /System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 0.0.0, current version 0.0.0)
```

Inspect shared libraries on Brew's binary:
```
❯ otool -L /opt/homebrew/bin/saml2aws
/opt/homebrew/bin/saml2aws:
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1858.112.0)
        /System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 1.0.0, current version 60158.100.133)
        /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
        /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 2113.40.126)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
```
Brew's has two additional libs in use: `IOKit.framework` and `AppKit.framework`. Is one of these where Keychain support comes from?

---

Now, if we check the same against the x86_64/AMD builds...

Download and extract the latest GH Release binary:
```
❯ wget -q https://github.com/Versent/saml2aws/releases/download/v2.36.0/saml2aws_2.36.0_Darwin_amd64.tar.gz

❯ tar xzvf saml2aws_2.36.0_Darwin_amd64.tar.gz
LICENSE.md
README.md
saml2aws
```

Check file type is x86_64
```
❯ file ./saml2aws
saml2aws: Mach-O 64-bit executable x86_64
```

Check shared libraries
```
❯ otool -L ./saml2aws
saml2aws:
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1856.105.0)
        /System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 1.0.0, current version 60157.60.19)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
        /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
        /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 2113.20.111)
```

---

Last check! Does using the AMD build via Rosetta work on M1? It sure does!

Using the same x86_64/AMD binary downloaded above. We see the appropriate log messages about Keychain, and once I grant the binary permissions to access they Keychain it logs in successfully.
```
❯ file ./saml2aws
saml2aws: Mach-O 64-bit executable x86_64

❯ ./saml2aws --skip-prompt login --force --verbose
DEBU[0000] Running                                       command=login
DEBU[0000] Check if creds exist.                         command=login
DEBU[0000] Expand                                        name=/Users/matt.klich/.aws/credentials pkg=awsconfig
DEBU[0000] resolveSymlink                                name=/Users/matt.klich/.aws/credentials pkg=awsconfig
DEBU[0000] ensureConfigExists                            filename=/Users/matt.klich/.aws/credentials pkg=awsconfig
Using IdP Account default to access REDACTED
DEBU[0000] Get credentials                               helper=osxkeychain serverURL="REDACTED"
DEBU[0000] Get credentials                               helper=osxkeychain user=REDACTED
DEBU[0000] Get credentials                               helper=osxkeychain serverURL="REDACTED"
```

**Comment by mapkon** [comment]
Fixed: #1243
